musicbot/bot.py

```diff
@@ -2546,9 +2546,22 @@ def check(message):
         return Response("Disconnected from `{0.name}`".format(server), delete_after=20)
 
     async def cmd_restart(self, channel):
+        """
+        Usage: 
+            {command_prefix}restart
+
+        Restarts the bot, making it leave all voice channels, this also opens a new command prompt, with any saved changes to files.
+        """
         await self.safe_send_message(channel, "\N{WAVING HAND SIGN}")
         await self.disconnect_all_voice_clients()
-        raise exceptions.RestartSignal()
+        from sys import platform as _platform
+        if _platform.startswith("linux"):
+	    p = subprocess.Popen(runbot.linux.sh, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        elif _platform == "darwin":
+            p = subprocess.Popen('runbot_osx.command', creationflags=subprocess.CREATE_NEW_CONSOLE)
+        else:
+            p = subprocess.Popen('runbot.bat', creationflags=subprocess.CREATE_NEW_CONSOLE)
```

Comment:
This isn't how subprocesses work. You're just going to end up creating multiple processes within the same process, and then you'll eventually run into memory errors.